csi/src/node/iscsiutil.rs

```diff
@@ -79,67 +82,97 @@ fn attach_disk(
         format!("/dev/disk/by-path/ip-{}-iscsi-{}-lun-{}", tp, iqn, lun);
     let iscsiadm = get_iscsiadm()?;
 
-    let target = format!("{},{} {}", tp, lun, iqn);
-
-    // If the device path exists then a previous invocation of this
-    // method has succeeded.
-    if wait_for_path_to_exist(device_path.as_str(), 1) {
-        trace!("path already exists!");
-        return Ok(iscsi_realpath(device_path));
-    }
+    static RE_SESSION: Lazy<regex::Regex> = Lazy::new(|| {
+        regex::Regex::new(
+            r"\s*(?P<tp>\d+.\d+.\d+.\d+:\d+),0\s*(?P<iqn>iqn.2019-05.io.openebs:nexus-.*)\s"
```

Comment:
Is there any way we can have this iqn defined in one place?

Reply:
Thats the subject for another change, it's not just the iqn prefix but also the nqn prefix.